---
title: 'HAP FFG RNA-seq: HDAC8 introduction (wt) comparison to inhibitor treatment and KO Top90 (updated to hg38_e100)'
author: "Adrian Arnel Lauber"
date: "15.04.2020"
output:
  html_document: 
    theme: cerulean
    highlight: monochrome
    toc: yes
    toc_float: yes
  word_document:
    toc: yes
  pdf_document: 
    toc: yes
  html_notebook: 
    toc: yes
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      include = TRUE,
                      warning=FALSE,
                      message=TRUE,
                      fig.align = 'center')
```

# Background & Summary
Comparison of the difference between introduction of inactive HDAC8 Mutant into wt and inhibitors. And additionally comparison to KO generation?

## Performed steps for DEseq2
* Load R libraries:

```{r imports, message=FALSE}
library("tidyverse")
library("DESeq2")
library("pheatmap")
library("writexl")
library("knitr")
```

```{r parallelization}
platform <- Sys.info()["sysname"]

if (platform == "Linux") {
  library("BiocParallel")
  register(MulticoreParam(2))
}
```

```{r functions}
# 1st function is the standard heatmap for wt background, returns gene count of input list
create_heatmap <- function(inp_list, n_genes, title, out_file, sort_decr = TRUE, clustered = TRUE, numbers = FALSE) {

  betas <- coef(dds)

  betas_sig <- betas[!(row.names(betas) %in% setdiff(rownames(betas), rownames(inp_list))), ]
  betas_sig <- betas_sig[match(rownames(inp_list), row.names(betas_sig)),]


  betascols <- c("intercept", "HDAC8KO", "JQ12 1µM", "MS275 1µM", "MS275 3µM", "HDAC8wt HDAC8Mut")
  attr(betas_sig, "dimnames") <- list(inp_list$gene_name, betascols)

  topGenes <- head(order(inp_list$log2FoldChange, decreasing = sort_decr), n_genes)
  mat <- betas_sig[topGenes, -1]
  
  thr <- 3
  colramp <- seq(from=-thr, to=thr, length=101)
  if (numbers == TRUE){
    thr <- 21
    colramp <- seq(from=-thr+12, to=thr-12, length=101)
  }
  
  mat[mat < -thr] <- -thr
  mat[mat > thr] <- thr

  col.order <- c("HDAC8wt HDAC8Mut", "MS275 1µM", "MS275 3µM", "JQ12 1µM", "HDAC8KO")
  mat <- mat[,col.order]

  pheatmap(mat, breaks = colramp, cluster_col = FALSE, cluster_row = clustered, #set treeheight_row = 0 to mask
           fontsize_col = 15, angle_col = 90, main = title, display_numbers = numbers,
           cellheight = 12, clustering_method = "complete",
           cellwidth = 50, legend = FALSE, fontsize = 10, fontsize_row = 10,
           filename = out_file)
  
  return(betas_sig)
}


# 2nd function creates the heatmap for comparing the genes of inhibitor treatment to others, returns gene count of input list
create_heatmap_inh <- function(inp_list, n_genes, title, out_file, sort_decr = TRUE, clustered = TRUE, numbers = FALSE) {

  betas <- coef(dds)

  betas_sig <- betas[!(row.names(betas) %in% setdiff(rownames(betas), rownames(inp_list))), ]
  betas_sig <- betas_sig[match(rownames(inp_list), row.names(betas_sig)),]


  betascols <- c("intercept", "HDAC8KO", "JQ12 1µM", "MS275 1µM", "MS275 3µM", "HDAC8wt HDAC8Mut")
  attr(betas_sig, "dimnames") <- list(inp_list$gene_name, betascols)

  topGenes <- head(order(inp_list$log2FoldChange, decreasing = sort_decr), n_genes)
  mat <- betas_sig[topGenes, -1]
  
  thr <- 3
  colramp <- seq(from=-thr, to=thr, length=101)
  if (numbers == TRUE){
    thr <- 21
    colramp <- seq(from=-thr+12, to=thr-12, length=101)
  }
  
  mat[mat < -thr] <- -thr
  mat[mat > thr] <- thr

  col.order <- c("MS275 1µM", "HDAC8wt HDAC8Mut", "HDAC8KO") #"MS275 3µM", "JQ12 1µM")
  mat <- mat[,col.order]

  pheatmap(mat, breaks = colramp, cluster_col = FALSE, cluster_row = clustered, #set treeheight_row = 0 to mask
           fontsize_col = 15, angle_col = 90, main = title, display_numbers = numbers,
           cellheight = 12, clustering_method = "complete", gaps_col = 3,
           cellwidth = 50, legend = FALSE, fontsize = 10, fontsize_row = 10,
           filename = out_file)
  
  return(betas_sig)
}
```

```{r variables}
#vector to collect the output results list names. Used for iteration over it and save the created lists as excel tables below.
out_lists <- vector()
```

* Read raw count matrix and coldata
  + Raw count matrix is provided by Michael Schuster of CEMM. 
  + nome: Homo sapiens genome assembly hg38 (December 2013)
  + Transcriptome: Homo sapiens transcriptome hg38_e100 (April 2020)
  + STAR aligner was used for mapping of reads.
  + The same RAW count numbers which where used in Schuster´s DEseq analysis are being read in here
  + Coldata: Additional information about each sample. they are stored in the Metadata slot of the DEseq object

```{r countmatrix}
cts_full <- read_tsv("rnaseq_deseq_global_counts_raw.tsv")
cts <- cts_full %>% select(gene_id, contains(c("HAP_143_", "HAP_1_", "HAP_2_", "HAP_3_", "HAP_JQ", "HAP_MS1_", "HAP_MS3", "HD8_KO"))) %>% column_to_rownames("gene_id")
cts_matrix <- as.matrix(cts)

coldata <- read.csv("rnaseq_deseq_inhibitor_samples.tsv", sep="\t", row.names=12, colClasses=c("NULL","NULL", "NULL", NA, NA, "NULL", NA, NA, NA, NA, "NULL", NA, "NULL", NA), stringsAsFactors = TRUE)


featureData <- cts_full %>% select(gene_version:location)

```

* Difference between sample metadata and count matrix samples?
  + The sample data table has to contain exactly the same samples in the same order as the count matrix table, where we chose already the samples of interest. If there is a difference, drop the following unused sample data.
```{r difference, echo=TRUE}
ctsc <- colnames(cts_matrix)
coldatac <- rownames(coldata)
setdiff(coldatac, ctsc)

coldata <- coldata[!(row.names(coldata) %in% setdiff(rownames(coldata),ctsc)), ]
```

* Do the sample data table and the count matrix now contain the same samples in the same order?
```{r sample_check, echo=TRUE}
(all(rownames(coldata) == colnames(cts)))
```

* Levels of Factors should of course also be adjusted to our "sub-matrix"

```{r correct_factors}
coldata$genotype <- factor(coldata$genotype, levels = c("wt", "HDAC8ko"))
coldata$expression <- factor(coldata$expression, levels = c("native", "iHDAC8"))
coldata$group <- factor(coldata$group, levels = c("wt_native", "HDAC8ko_native", "wt_JQ12_1uM_24h", "wt_MS275_1uM_24h",
                                                  "wt_MS275_3uM_24h", "wt_inactive_HDAC8"))
```

* Create dds object + add metadata

```{r dds_object_initialization}
dds <- DESeqDataSetFromMatrix(countData = cts_matrix,
                              colData = coldata,
                              design = ~ group)

mcols(dds) <- DataFrame(mcols(dds), featureData)
```

* relevel factors to set the reference

```{r relevel_group_factor}
dds$group <- factor(dds$group, levels = c("wt_native", "HDAC8ko_native", "wt_JQ12_1uM_24h", "wt_MS275_1uM_24h",
                                          "wt_MS275_3uM_24h", "wt_inactive_HDAC8"))
```

* Pre-filter the low raw count rows
  + Here we can define the threshold for the sum of raw counts per row, so that rows (=genes) with a generally low count can be dropped. This threshold can also depend on the number of samples!

```{r cutoff, echo=TRUE}
keep <- rowSums(counts(dds)) > 1
dds <- dds[keep,]
```

* Deseq with LRT

```{r deseq}
if (platform == "Linux") {
  dds <- DESeq(dds, test="LRT", reduced=~1, parallel=TRUE)
} else {
  dds <- DESeq(dds, test="LRT", reduced=~1)
}

ci_vs_ctrl_all <- results(dds)
```

* Quality control
```{r quality_check}
summary(ci_vs_ctrl_all)
options(scipen = 5)

plotMA(ci_vs_ctrl_all, ylim = c(-10,10), main = "group HDAC8wt HDAC8Mut vs wt native \n rowsum > 5")

options(scipen = 0)
```

## Adding gene names/Types and chose significant genes for Heatmaps
For this task additional result tables are being generated. So I can chose the genes which fulfill the conditions.

* add gene names to main gene set with log2foldChanges of ci introduction relative to ctrl
```{r main_set_names}
ci_vs_ctrl_all$gene_name <- mcols(dds)$gene_name
ci_vs_ctrl_all$gene_biotype <- mcols(dds)$gene_biotype
out_lists <- c(out_lists, "ci_vs_ctrl_all")
```

* Choose significant genes from main gene set and subset into upregulated, downregulated and unchanged genes (ci introduction relative to ctrl)
```{r significants}
ci_vs_ctrl <- subset(ci_vs_ctrl_all, padj < 0.1)
out_lists <- c(out_lists, "ci_vs_ctrl")

ci_vs_ctrl_up <- subset(ci_vs_ctrl_all, log2FoldChange > 1 & padj < 0.1)
ci_vs_ctrl_up <- ci_vs_ctrl_up[order(ci_vs_ctrl_up$log2FoldChange, decreasing = TRUE),]
out_lists <- c(out_lists, "ci_vs_ctrl_up")

ci_vs_ctrl_unch <- subset(ci_vs_ctrl_all, log2FoldChange < 1 & log2FoldChange > -1 & padj < 0.1)
ci_vs_ctrl_unch <- ci_vs_ctrl_unch[order(ci_vs_ctrl_unch$log2FoldChange, decreasing = TRUE),]
out_lists <- c(out_lists, "ci_vs_ctrl_unch")

ci_vs_ctrl_down <- subset(ci_vs_ctrl_all, log2FoldChange < -1 & padj < 0.1)
ci_vs_ctrl_down <- ci_vs_ctrl_down[order(ci_vs_ctrl_down$log2FoldChange, decreasing = TRUE),]
out_lists <- c(out_lists, "ci_vs_ctrl_down")
```

* Extract list of significant genes which are deregulated in KO (relative to ctrl)
```{r KO_significants}
KO_vs_ctrl_all <- results(dds, contrast = c("group", "HDAC8ko_native", "wt_native"))
KO_vs_ctrl_all$gene_name <- mcols(dds)$gene_name
KO_vs_ctrl_all$gene_biotype <- mcols(dds)$gene_biotype
out_lists <- c(out_lists, "KO_vs_ctrl_all")
```

* Group deregulated genes in KO into upregulated and unchanged/downregulated and reversed
```{r groups_KO}
KO_vs_ctrl <- subset(KO_vs_ctrl_all, padj < 0.1)
out_lists <- c(out_lists, "KO_vs_ctrl")

KO_vs_ctrl_up <- subset(KO_vs_ctrl_all, log2FoldChange > 1 & padj < 0.1)
KO_vs_ctrl_up <- KO_vs_ctrl_up[order(KO_vs_ctrl_up$log2FoldChange, decreasing = TRUE),]
out_lists <- c(out_lists, "KO_vs_ctrl_up")

KO_vs_ctrl_down <- subset(KO_vs_ctrl_all, log2FoldChange < -1 & padj < 0.1)
KO_vs_ctrl_down <- KO_vs_ctrl_down[order(KO_vs_ctrl_down$log2FoldChange, decreasing = FALSE),]
out_lists <- c(out_lists, "KO_vs_ctrl_down")

KO_vs_ctrl_unchdown <- subset(KO_vs_ctrl_all, log2FoldChange < 1 & padj < 0.1)
KO_vs_ctrl_unchdown <- KO_vs_ctrl_unchdown[order(KO_vs_ctrl_unchdown$log2FoldChange, decreasing = TRUE),]
out_lists <- c(out_lists, "KO_vs_ctrl_unchdown")

KO_vs_ctrl_unchup <- subset(KO_vs_ctrl_all, log2FoldChange > -1 & padj < 0.1)
KO_vs_ctrl_unchup <- KO_vs_ctrl_unchup[order(KO_vs_ctrl_unchup$log2FoldChange, decreasing = TRUE),]
out_lists <- c(out_lists, "KO_vs_ctrl_unchup")
```


* Group ci and KO lists into sublists
```{r KO_sublists}
ci_up_KO_up_vs_ctrl <- ci_vs_ctrl_up[!(rownames(ci_vs_ctrl_up) %in%
                               setdiff(rownames(ci_vs_ctrl_up),rownames(KO_vs_ctrl_up))), ]
out_lists <- c(out_lists, "ci_up_KO_up_vs_ctrl")

ci_down_KO_down_vs_ctrl <- ci_vs_ctrl_down[!(rownames(ci_vs_ctrl_down) %in%
                               setdiff(rownames(ci_vs_ctrl_down),rownames(KO_vs_ctrl_down))), ]
out_lists <- c(out_lists, "ci_down_KO_down_vs_ctrl")

ci_up_KO_unchdown_vs_ctrl <- ci_vs_ctrl_up[!(rownames(ci_vs_ctrl_up) %in%
                               setdiff(rownames(ci_vs_ctrl_up),rownames(KO_vs_ctrl_unchdown))), ]
out_lists <- c(out_lists, "ci_up_KO_unchdown_vs_ctrl")

ci_down_KO_unchup_vs_ctrl <- ci_vs_ctrl_down[!(rownames(ci_vs_ctrl_down) %in%
                               setdiff(rownames(ci_vs_ctrl_down),rownames(KO_vs_ctrl_unchup))), ]
out_lists <- c(out_lists, "ci_down_KO_unchup_vs_ctrl")
```


* Create the list (extract) of deregulation for ms275_1uM
```{r ms275_significants}
ms275_vs_ctrl_all <- results(dds, contrast = c("group", "wt_MS275_1uM_24h", "wt_native"))
ms275_vs_ctrl_all$gene_name <- mcols(dds)$gene_name
ms275_vs_ctrl_all$gene_biotype <- mcols(dds)$gene_biotype
out_lists <- c(out_lists, "ms275_vs_ctrl_all")
```

* Group the ms275 genes into up- and downregulated genes
```{r ms275_groups}
ms275_vs_ctrl_up <- subset(ms275_vs_ctrl_all, log2FoldChange > 1 & padj < 0.1)
ms275_vs_ctrl_up <- ms275_vs_ctrl_up[order(ms275_vs_ctrl_up$log2FoldChange, decreasing = TRUE),]
out_lists <- c(out_lists, "ms275_vs_ctrl_up")

ms275_vs_ctrl_down <- subset(ms275_vs_ctrl_all, log2FoldChange < -1 & padj < 0.1)
ms275_vs_ctrl_down <- ms275_vs_ctrl_down[order(ms275_vs_ctrl_down$log2FoldChange, decreasing = FALSE),]
out_lists <- c(out_lists, "ms275_vs_ctrl_down")
```


* Save all created lists additionally as Excel Sheets and count gene numbers & calculate ratios.
```{r create_lists}
out_lists2 = out_lists
comparison = vector()
ratio = vector()
out_lists_counts = vector()

for (listname in out_lists) {
  write_xlsx(rownames_to_column(as.data.frame(get(listname)), "gene_id"),
           path = paste0("20210415_HDACwt_ci_KO_inh_report/HDAC8_", listname, ".xlsx"))
}

#count and calculate ratios of gene numbers between all lists
for (listname1 in out_lists) {
  for (listname2 in out_lists2) {
    comparison = c(comparison, paste0(listname1, "_from_", listname2))
    
    list1 = get(listname1) #This is unecessary, should be moved one layer above
    count1 = length(list1$gene_name)
    list2 = get(listname2)
    count2 = length(list2$gene_name)
    
    current_ratio = count1/count2*100
    ratio = c(ratio, current_ratio)
  }
  out_lists_counts = c(out_lists_counts, count1) #This was added to see the gene numbers in the lists
}

all_ratios = data.frame(comparison, ratio)
usefull_ratios = all_ratios[all_ratios$ratio <= 100, ]
write_xlsx(usefull_ratios, path = paste0("20210415_HDACwt_ci_KO_inh_report/HDAC8_ratios.xlsx"))
all_ratios = column_to_rownames(all_ratios, "comparison")

gene_numbers = data.frame(out_lists, out_lists_counts)
write_xlsx(gene_numbers, path = paste0("20210415_HDACwt_ci_KO_inh_report/HDAC8_GeneCounts.xlsx"))
gene_numbers = column_to_rownames(gene_numbers, "out_lists")
```




# Gene counts (general numbers and ratios)

The significant genes were chosen according to whether they experience deregulation in any cell type or treatmen.
```{r general_numbers_1}
print(paste0("Total gene number used as Input for DEseq: ", gene_numbers["ci_vs_ctrl_all",]))
print(paste0("Significant genes: ", gene_numbers["ci_vs_ctrl",]))
```



# Heatmaps of HDAC8Mut introduction
Get the FoldChange Estimates from dds object and collapse it to have only significant genes in the list.
Since the object betas has the gene ids as Attribute, change them to gene names in order to see the names in the Heatmap. Create Heatmaps according to the grouping above.
This part contains the heatmaps which rank the genes according to regulation in the HDAC8Mut introduction.

## i) Top genes in HDAC8wt HDAC8Mut (general up&down)
### complete Heatmap
```{r gene_heatmap_up, out.height = "600px", out.width = "70%"}
res_table <- create_heatmap(ci_vs_ctrl_up, 10000, "upregulated upon ci intro", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up.pdf")
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up.pdf")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_vs_ctrl_up_from_ci_vs_ctrl_all",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up.xlsx)

```{r gene_heatmap_down, out.height = "600px", out.width = "70%"}
res_table <- create_heatmap(ci_vs_ctrl_down, 10000, "downregulated upon ci intro",
                            "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down.pdf", sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down.pdf")
message("genes: ", length(rownames(res_table)))
message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
message("percentage from all deregulated genes: ", all_ratios["ci_vs_ctrl_down_from_ci_vs_ctrl_all",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down.xlsx)

### Top90 with numbers
```{r gene_heatmap_up90n, out.width = "50%"}
res_table <- create_heatmap(ci_vs_ctrl_up, 90, "upregulated upon ci intro", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90n.png",
                            numbers = TRUE, clustered = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90n.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90n.png")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_vs_ctrl_up_from_ci_vs_ctrl_all",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up.xlsx)

```{r gene_heatmap_down90n, out.width = "50%"}
res_table <- create_heatmap(ci_vs_ctrl_down, 90, "downregulated upon ci intro",
                            "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90n.png", sort_decr = FALSE, numbers = TRUE, clustered = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90n.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90n.png")
message("genes: ", length(rownames(res_table)))
message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
message("percentage from all deregulated genes: ", all_ratios["ci_vs_ctrl_down_from_ci_vs_ctrl_all",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down.xlsx)

### Top90 clustered
```{r gene_heatmap_up90, out.width = "50%"}
res_table <- create_heatmap(ci_vs_ctrl_up, 90, "upregulated upon ci intro", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90.png")
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90.png")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_vs_ctrl_up_from_ci_vs_ctrl_all",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up.xlsx)

```{r gene_heatmap_down90, out.width = "50%"}
res_table <- create_heatmap(ci_vs_ctrl_down, 90, "downregulated upon ci intro",
                            "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90.png", sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90.png")
message("genes: ", length(rownames(res_table)))
message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
message("percentage from all deregulated genes: ", all_ratios["ci_vs_ctrl_down_from_ci_vs_ctrl_all",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down.xlsx)

### Top90 ranked
```{r gene_heatmap_up90rank, out.width = "50%"}
res_table <- create_heatmap(ci_vs_ctrl_up, 90, "upregulated upon ci intro", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90rank.png",
                            clustered = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90rank.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up90rank.png")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_vs_ctrl_up_from_ci_vs_ctrl_all",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up.xlsx)

```{r gene_heatmap_down90rank, out.width = "50%"}
res_table <- create_heatmap(ci_vs_ctrl_down, 90, "downregulated upon ci intro",
                            "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90rank.png", sort_decr = FALSE, clustered = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90rank.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down90rank.png")
message("genes: ", length(rownames(res_table)))
message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
message("percentage from all deregulated genes: ", all_ratios["ci_vs_ctrl_down_from_ci_vs_ctrl_all",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down.xlsx)


## ii) Top genes (up) in HDAC8wt HDAC8Mut when not upregulated in KO
### complete heatmap
```{r ci_up_KO_unchdown_vs_ctrl, out.height = "600px", out.width = "70%"}
res_table <- create_heatmap(ci_up_KO_unchdown_vs_ctrl, 10000, "upregulated upon ci intro \n no upregulation in KO", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_unchdown.pdf")
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_unchdown.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_unchdown.pdf")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_up_KO_unchdown_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_unchdown.xlsx)

### Top90 clustered
```{r ci_up_KO_unchdown_vs_ctrl90, out.width = "50%"}
res_table <- create_heatmap(ci_up_KO_unchdown_vs_ctrl, 90, "upregulated upon ci intro \n no upregulation in KO", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_unchdown90.png")
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_unchdown90.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_unchdown90.png")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_up_KO_unchdown_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_unchdown90.xlsx)

## ii) Top genes (down) in HDAC8wt HDAC8Mut when not downregulated in KO
### complete heatmap
```{r ci_down_KO_unchup_vs_ctrl, out.height = "600px", out.width = "70%"}
res_table <- create_heatmap(ci_down_KO_unchup_vs_ctrl, 10000, "downregulated upon ci intro \n no upregulation in KO", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_down_KO_unchup.pdf", sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_down_KO_unchup.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_down_KO_unchup.pdf")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_down_KO_unchup_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_down_KO_unchup.xlsx)

### Top90 clustered
```{r ci_down_KO_unchup_vs_ctrl90, out.width = "50%"}
res_table <- create_heatmap(ci_down_KO_unchup_vs_ctrl, 90, "downregulated upon ci intro \n no upregulation in KO", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_down_KO_unchup90.png", sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_down_KO_unchup90.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_down_KO_unchup90.png")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_down_KO_unchup_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_down_KO_unchup90.xlsx)

## iii) Top genes in HDAC8wt HDAC8Mut which are also upregulated in KO
### complete heatmap
```{r ci_up_KO_up_vs_ctrl, out.height = "600px", out.width = "70%"}
res_table <- create_heatmap(ci_up_KO_up_vs_ctrl, 10000, "upregulated upon ci intro \n upregulated in KO", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_up.pdf")
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_up.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_up.pdf")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_up_KO_up_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_up.xlsx)

### Top90 clustered
```{r ci_up_KO_up_vs_ctrl90, out.width = "50%"}
res_table <- create_heatmap(ci_up_KO_up_vs_ctrl, 90, "upregulated upon ci intro \n upregulated in KO", "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_up90.png")
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_up90.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_up90.png")
message("genes: ", length(rownames(res_table)))
message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
message("percentage from all deregulated genes: ", all_ratios["ci_up_KO_up_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ci_up_KO_up90.xlsx)

## iii) Top genes in HDAC8wt HDAC8Mut which are also downregulated in KO
### complete heatmap
```{r ci_down_KO_down_vs_ctrl, out.height = "600px", out.width = "70%"}
res_table <- create_heatmap(ci_down_KO_down_vs_ctrl, 10000, "downregulated downon ci intro \n downregulated in KO", "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down_KO_down.pdf", sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down_KO_down.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down_KO_down.pdf")
message("genes: ", length(rownames(res_table)))
message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
message("percentage from all deregulated genes: ", all_ratios["ci_down_KO_down_vs_ctrl_from_ci_vs_ctrl_down",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down_KO_down.xlsx)

### Top90 clustered
```{r ci_down_KO_down_vs_ctrl90, out.width = "50%"}
res_table <- create_heatmap(ci_down_KO_down_vs_ctrl, 90, "downregulated downon ci intro \n downregulated in KO", "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down_KO_down90.png", sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down_KO_down90.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down_KO_down90.png")
message("genes: ", length(rownames(res_table)))
message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
message("percentage from all deregulated genes: ", all_ratios["ci_down_KO_down_vs_ctrl_from_ci_vs_ctrl_down",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ci_down_KO_down90.xlsx)


# Heatmaps of inhibitor Treatment
Same as above but here the genes are ranked according to the deregulation by inhibitor treatment.

## i) general (up)
### complete heatmap
```{r ms275_up, out.height = "600px", out.width = "70%"}
res_table <- create_heatmap_inh(ms275_vs_ctrl_up, 10000, "upregulated by MS275 treatment", "20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up.pdf", clustered = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up.pdf")
message("genes: ", length(rownames(res_table)))
#message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
#message("percentage from all deregulated genes: ", all_ratios["ci_up_KO_up_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ms275_vs_ctrl_up.xlsx)

### Top90 clustered
```{r ms275_up90, out.width = "50%"}
res_table <- create_heatmap_inh(ms275_vs_ctrl_up, 90, "upregulated by MS275 treatment", "20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up90.png")
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up90.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up90.png")
message("genes: ", length(rownames(res_table)))
#message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
#message("percentage from all deregulated genes: ", all_ratios["ci_up_KO_up_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ms275_vs_ctrl_up90.xlsx)

### Top90 with numbers
```{r ms275_up90n, out.width = "50%"}
res_table <- create_heatmap_inh(ms275_vs_ctrl_up, 90, "upregulated by MS275 treatment", "20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up90n.png", clustered = FALSE, numbers = TRUE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up90n.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ms275_up/HDAC8_ms275_vs_ctrl_up90n.png")
message("genes: ", length(rownames(res_table)))
#message("Total upregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_up",])
#message("percentage from all deregulated genes: ", all_ratios["ci_up_KO_up_vs_ctrl_from_ci_vs_ctrl_up",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_up/HDAC8_ms275_vs_ctrl_up90n.xlsx)


## i) general (down)
### complete heatmap
```{r ms275_down, out.height = "600px", out.width = "70%"}
res_table <- create_heatmap_inh(ms275_vs_ctrl_down, 10000, "downregulated by MS275 treatment", "20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down.pdf", clustered = FALSE, sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down.pdf")
message("genes: ", length(rownames(res_table)))
#message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
#message("percentage from all deregulated genes: ", all_ratios["ci_down_KO_down_vs_ctrl_from_ci_vs_ctrl_down",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ms275_vs_ctrl_down.xlsx)

### Top90 clustered
```{r ms275_down90, out.width = "50%"}
res_table <- create_heatmap_inh(ms275_vs_ctrl_down, 90, "downregulated by MS275 treatment", "20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down90.png", sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down90.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down90.png")
message("genes: ", length(rownames(res_table)))
#message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
#message("percentage from all deregulated genes: ", all_ratios["ci_down_KO_down_vs_ctrl_from_ci_vs_ctrl_down",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ms275_vs_ctrl_down90.xlsx)

### Top90 with numbers
```{r ms275_down90n, out.width = "50%"}
res_table <- create_heatmap_inh(ms275_vs_ctrl_down, 90, "downregulated by MS275 treatment", "20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down90n.png", clustered = FALSE, numbers = TRUE, sort_decr = FALSE)
write_xlsx(rownames_to_column(as.data.frame(res_table), "gene_name"),
           path = "20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down90n.xlsx")
include_graphics("20210415_HDACwt_ci_KO_inh_report/ms275_down/HDAC8_ms275_vs_ctrl_down90n.png")
message("genes: ", length(rownames(res_table)))
#message("Total downregulated in HDAC8wt HDAC8Mut: ", gene_numbers["ci_vs_ctrl_down",])
#message("percentage from all deregulated genes: ", all_ratios["ci_down_KO_down_vs_ctrl_from_ci_vs_ctrl_down",],"%")
```
[download Excel Table](20210415_HDACwt_ci_KO_inh_report/ci_down/HDAC8_ms275_vs_ctrl_down90n.xlsx)

# R status
```{r}
sessionInfo()
```
